sipyco/test/test_pc_rpc.py

```diff
@@ -48,13 +48,22 @@ def _blocking_echo(self, target):
             self.assertEqual(test_object, test_object_back)
             with self.assertRaises(AttributeError):
                 remote.non_existing_method
-            remote.terminate()
+            if die_using_sys_exit:
+                # If the server dies and just drops the connection, we
+                # expect a client-side error due to lack of data.
+                with self.assertRaises(SyntaxError):
```

Comment:
Hmm, not such a nice exception. Maybe ``SystemExit`` should in fact be caught, and handled specifically?

Reply:
I'd say that's more a client-side issue – the server might disappear spontaneously for other reasons too.

This whole situation is more of a mess than I had anticipated in another way as well: I just found out that depending on the event loop implementation the default exception handler might still just kill the server coroutine on receiving the `SystemExit` exception rather than the whole program. In typical RPC servers (and the test case), this will indeed terminate the process, but that's still not very clean.

There is a Python bug for this, but of course this is less than ideal.

Basically, there are two relevant scenarios:

 1) Some legacy code actually calls `sys.exit` and expects the process to die (and e.g. left some DLL in an invalid state, etc.). Minimizing surprises here was the main goal for me.

 2) Green-field code would like to terminate the server because of an error that is not easy to recover from. This is often a sensible strategy, as the controller manager (or a systemd unit, or some other supervisor) will shortly restart the process. Committing suicide is then a robust strategy (vs. dealing with releasing and reacquiring resources as needed) if the short blip in availability can be tolerated.

For the first, we depend on `SystemExit`, but don't necessarily need a very clean response, as this should be rare. A clean solution for the second would be nice, but we could also come up with some other mechanism of signaling this to `Server`.

Reply:
> depending on the event loop implementation the default exception handler might still just kill the server coroutine on receiving the `SystemExit` exception rather than the whole program

Catching it and terminating the server coroutine should clean up that behavior, no?